src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -780,7 +783,11 @@ private bool InferNamedPipesInformation()
             if (_dataSourceAfterTrimmingProtocol.StartsWith(PipeBeginning) || ConnectionProtocol == Protocol.NP)
             {
                 // If the data source is "np:servername"
-                if (!_dataSourceAfterTrimmingProtocol.Contains(BackSlashSeparator))
+#if netcoreapp
+                if (!_dataSourceAfterTrimmingProtocol.Contains(BackSlashSeparator)) // string.Contains(char) is .NetCore2.1+ specific
+#else
+                if (!System.Linq.Enumerable.Contains(_dataSourceAfterTrimmingProtocol, BackSlashSeparator))
```

Comment:
Note explicit call to extension. At least we know explicitly that the perf will suck now :-)

Reply:
Don't feel right about that. Reverting and using `#if` instead